subprojects/parseq-guava-interop/src/main/java/com/linkedin/parseq/guava/ListenableFutureUtil.java

```diff
@@ -38,9 +43,13 @@ public boolean cancel(Exception rootReason) {
 
             return super.cancel(rootReason) && future.cancel(true);
           }
+
+          @Override
+          protected Promise<? extends T> run(Context context) throws Throwable {
+            return promise;
```

Comment:
why cannot we instantiate promise here inside `run`, and also move the code of callbackRunnable inside here?

Reply:
the task can get canceled during the callback. Therefore the callback should be in a context that can refer to that task.

Reply:
thee task can get cancelled during the callback. Therefore the callback should be in a context that refer to the task.

Reply:
thanks for the explanation. got it.